src/wallet/wallet.cpp

```diff
@@ -1947,7 +1947,9 @@ CAmount CWalletTx::GetAvailableCredit(bool fUseCache, const isminefilter& filter
 
     if (cache) {
         *cache = nCredit;
-        *cache_used = true;
+        if (cache_used) {
```

Comment:
If `cache` is set then `cache_used` is also set. Maybe `assert(cache_used)` instead?